yarn/src/main/scala/org/apache/spark/deploy/yarn/LocalityPreferredContainerPlacementStrategy.scala

```diff
@@ -179,4 +196,26 @@ private[yarn] class LocalityPreferredContainerPlacementStrategy(
       (host, math.max(0, (expectedCount - existedCount).ceil.toInt))
     }
   }
+
+  /**
+   * Calculate the host to possible number of containers for pending allocated containers.
+   * @param localityMatchedPendingAllocations A sequence of pending container request which
+   *                                          matches the localities of current required tasks.
+   * @return a Map with hostname as key and possible number of containers on this host as value
```

Comment:
This confused me a little bit; the "possible number of containers" would be just the raw count stored in `localityMatchedPendingAllocations`, no? Instead, the method is returning some kind of ratio. It would be nice to clarify that here.